osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/GetPowerQualityProfileSelectiveAccessHandlerTest.java

```diff
@@ -62,41 +76,63 @@ public void testHandlePrivateProfileSelectiveAccess() throws ProtocolAdapterExce
             any(String.class),
             any(AttributeAddress.class)))
         .thenReturn(
-            this.createCaptureObjects(),
+            this.createPrivateCaptureObjects(),
             this.createProfileEntries(),
-            this.createCaptureObjectsProfile2(),
+            this.createPrivateCaptureObjectsProfile2(),
             this.createProfileEntries());
 
-    when(this.dlmsHelper.readLogicalName(any(DataObject.class), any(String.class)))
-        .thenCallRealMethod();
-    when(this.dlmsHelper.readObjectDefinition(any(DataObject.class), any(String.class)))
-        .thenCallRealMethod();
-    when(this.dlmsHelper.readLongNotNull(any(DataObject.class), any(String.class)))
-        .thenCallRealMethod();
-    when(this.dlmsHelper.readLong(any(DataObject.class), any(String.class))).thenCallRealMethod();
-    when(this.dlmsHelper.convertDataObjectToDateTime(any(DataObject.class))).thenCallRealMethod();
-    when(this.dlmsHelper.fromDateTimeValue(any())).thenCallRealMethod();
-    when(this.dlmsHelper.getClockDefinition()).thenCallRealMethod();
-
     final GetPowerQualityProfileSelectiveAccessHandler handler =
         new GetPowerQualityProfileSelectiveAccessHandler(this.dlmsHelper);
 
-    // EXECUTE
-
     final GetPowerQualityProfileResponseDto responseDto =
         handler.handle(this.conn, this.dlmsDevice, requestDto);
 
-    // ASSERT
+    assertThat(responseDto.getPowerQualityProfileResponseDatas()).hasSize(2);
+    assertThat(responseDto.getPowerQualityProfileResponseDatas().get(0).getCaptureObjects())
+        .hasSize(3);
+    assertThat(responseDto.getPowerQualityProfileResponseDatas().get(0).getProfileEntries())
+        .hasSize(4);
 
-    assertThat(responseDto.getPowerQualityProfileResponseDatas().size()).isEqualTo(2);
-    assertThat(responseDto.getPowerQualityProfileResponseDatas().get(0).getCaptureObjects().size())
-        .isEqualTo(3);
-    assertThat(responseDto.getPowerQualityProfileResponseDatas().get(0).getProfileEntries().size())
-        .isEqualTo(4);
+    for (final ProfileEntryDto profileEntryDto :
+        responseDto.getPowerQualityProfileResponseDatas().get(0).getProfileEntries()) {
+      assertThat(profileEntryDto.getProfileEntryValues()).hasSize(3);
+    }
+  }
+
+  @Test
+  void testHandlePublicProfileSelectiveAccess() throws ProtocolAdapterException {
+
+    final GetPowerQualityProfileRequestDataDto requestDto =
+        new GetPowerQualityProfileRequestDataDto(
+            "PUBLIC",
+            Date.from(Instant.now().minus(2, ChronoUnit.DAYS)),
+            new Date(),
+            new ArrayList<>());
+
+    when(this.dlmsHelper.getAndCheck(
+            any(DlmsConnectionManager.class),
+            any(DlmsDevice.class),
+            any(String.class),
+            any(AttributeAddress.class)))
```

Comment:
`Any` allows any list of adresses. If there is a duplicate clock entry, then this is not caught.

Reply:
We have discussed this and decided to take this into account when refactoring the dataobjects.